test/models.spec.ts

```diff
@@ -529,12 +529,93 @@ describe("Unit | Filters", function () {
     });
   });
 
+  describe('validateFiltersContainer', function () {
+    it("validator should return errors if object does NOT validate against filtersContainer schema", function () {
+      // Arrange
+      const malformedFiltersContainer = {
+        abc: '123'
+      };
+
+      // Act
+      const errors = models.validateFiltersContainer(malformedFiltersContainer);
+
+      // Assert
+      expect(errors).toBeDefined();
+    });
+
+    it("validator should return no errors if object does validate against filtersContainer schema", function () {
+      // Arrange
+      const things: (string | number)[] = [
+        "a",
+        3234
+      ];
+
+      const basicFilter: models.IBasicFilter = {
+        $schema: "a",
+        target: {
+          table: "table",
+          column: "column"
+        },
+        operator: "In",
+        values: [
+          "A",
+          "B"
+        ]
+      };
+      const advancedFilter: models.IAdvancedFilter = {
+        $schema: "a",
+        target: {
```

Comment:
in the product filters have levels - page level filter, report level filter and visual level filters. So I think **level** would be a bit more appropriate than scope. At the same time just ILevel instead of ITarget sounds strange. 

Maybe we should rename as follow: ITarget -> ILevelFilter, IPageTarget->IPageLevelFilter, IVisualTarget->IVisualLevelFilter.